packages/workspaces-extension/src/index.ts

```diff
@@ -46,13 +50,84 @@ const workspacesMenu: JupyterFrontEndPlugin<void> = {
   }
 };
 
+const WORKSPACE_INDICATOR_PLUGIN_ID =
+  '@jupyterlab/workspaces-extension:indicator';
+const WORKSPACE_INDICATOR_COMMAND_ID = 'workspace-indicator:toggle';
+
+/**
+ * The extension providing workspace indicator at topbar
+ */
+const workspacesIndicator: JupyterFrontEndPlugin<void> = {
+  id: WORKSPACE_INDICATOR_PLUGIN_ID,
+  description: 'Adds a workspace indicator element at topbar',
+  requires: [
+    IWorkspacesModel,
+    IWorkspaceCommands,
+    IWindowResolver,
+    ITranslator,
+    ISettingRegistry
+  ],
+  autoStart: true,
+  activate: async (
+    app: JupyterFrontEnd,
+    model: IWorkspacesModel,
+    commands: IWorkspaceCommands,
+    resolver: IWindowResolver,
+    translator: ITranslator,
+    registry: ISettingRegistry
+  ) => {
+    const trans = translator.load('jupyterlab');
+    const openWorkspace = async (workspace: string) => {
+      await app.commands.execute(commands.open, { workspace: workspace });
+    };
+    const workspaceSelector = new WorkspaceSelectorWidget({
+      currentWorkspace: resolver.name,
+      identifiers: model.identifiers,
+      openWorkspace: openWorkspace,
+      model: model,
+      translator: translator
+    });
+    app.shell.add(workspaceSelector, 'top', { rank: 1000 });
```

Comment:
Wondering if this workspace selector item should be specified as a `Topbar` toolbar item?

https://github.com/jupyterlab/jupyterlab/blob/1f1a645d840d28ab7a85b751cf1df7c370a00254/packages/application-extension/schema/top-bar.json#L4-L12

This could let users specify where they would like to the selector to be displayed by changing the `rank`?